components/base/kinematicbase/execution.go

```diff
@@ -219,16 +220,16 @@ func (ptgk *ptgBaseKinematics) trajectoryArcSteps(
 
 	finalSteps := []arcStep{}
 	timeStep := 0.
-	curDist := 0.
-	curInputs := []referenceframe.Input{
+	curDist := inputs[startDistanceAlongTrajectoryIndex].Value
+	startInputs := []referenceframe.Input{
 		inputs[ptgIndex],
 		inputs[trajectoryAlphaWithinPTG],
-		{curDist},
-		inputs[endDistanceAlongTrajectoryIndex],
+		inputs[startDistanceAlongTrajectoryIndex],
```

Comment:
same question here, why is it ok to use `startDistanceAlongTrajectoryIndex` and not use `endDistanceAlongTrajectoryIndex`. Wouldn't a transform off this input just be zero?

Reply:
As above; this is not necessarily zero.

Reply:
Ah, I misread your earlier comment. You are correct. A transform off of that input would give us a zero pose.

This is intentional. At the start of a segment, we have not moved from the start position, so locating the base with Compose(startPosition, Transform(inputs)) should yield the start position.